tests/ecosystem/deployment/test_deployment.py

```diff
@@ -15,9 +16,139 @@ def test_deployment():
+    pod = ocp.OCP(
+        kind=constants.POD, namespace=namespace
+    )
+    timeout = 0
+    # ocs-operator
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector='name=ocs-operator',
+        timeout=timeout
+    )
+    # storagecluster-operator
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector=f'app={storage_cluster_name}-operator',
+        timeout=timeout
+    )
+    # nooba
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector='app=noobaa',
+        resource_count=2,
+        timeout=timeout
+    )
+    # local-storage-operator
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector='name=local-storage-operator',
+        timeout=timeout
+    )
+    # mons
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector=f'app={storage_cluster_name}-mon',
+        resource_count=3,
+        timeout=timeout
+    )
+    # csi-cephfsplugin
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector='app=csi-cephfsplugin',
+        resource_count=3
+    )
+    # csi-cephfsplugin-provisioner
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector='app=csi-cephfsplugin-provisioner',
+        resource_count=2,
+        timeout=timeout
+    )
+    # csi-rbdplugin
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector='app=csi-rbdplugin',
+        resource_count=3,
+        timeout=timeout
+    )
+    # csi-rbdplugin-profisioner
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector='app=csi-rbdplugin-provisioner',
+        resource_count=2,
+        timeout=timeout
+    )
+    # osds
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector=f'app={storage_cluster_name}-osd',
+        resource_count=3,
+        timeout=timeout
+    )
+    # mgr
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector=f'app={storage_cluster_name}-mgr',
+        timeout=timeout
+    )
+    # mds
+    assert pod.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector=f'app={storage_cluster_name}-mds',
+        resource_count=2,
+        timeout=timeout
+    )
+
+    # Verify StorageClasses (1 ceph-fs, 1 ceph-rbd)
+    storage_class = ocp.OCP(
+        kind=constants.STORAGECLASS, namespace=namespace
+    )
+    required_storage_classes = {
+        f'{storage_cluster_name}-cephfs',
+        f'{storage_cluster_name}-ceph-rbd'
+    }
+    storage_classes = storage_class.get()
+    storage_class_names = {
+        item['metadata']['name'] for item in storage_classes['items']
+    }
+    assert required_storage_classes.issubset(storage_class_names)
```

Comment:
Do feel we can add ceph health check as well here ? OR is it already covered elsewhere ?

Reply:
All of the verification steps were taken from the [doc](https://docs.google.com/document/d/1iQfIGR1N_Uezzo0gs-pSDFQ1q_oANWZGGIuslhmdQqg/edit). If we want to add additional checks here we can, but this PR is specifically for matching what is outlined in the doc.

Reply:
I thought about it again and the doc actually says to check ceph health from the dashboard (which we aren't doing here) so in the next update I'll add a ceph health check.

Reply:
Done